bot/exts/moderation/silence.py

```diff
@@ -188,14 +323,21 @@ async def _unsilence(self, channel: TextChannel) -> bool:
             log.info(f"Tried to unsilence channel #{channel} ({channel.id}) but the channel was not silenced.")
             return False
 
-        overwrite = channel.overwrites_for(self._verified_role)
+        if isinstance(channel, TextChannel):
+            overwrite = channel.overwrites_for(self._verified_msg_role)
+        else:
+            overwrite = channel.overwrites_for(self._verified_voice_role)
+
         if prev_overwrites is None:
             log.info(f"Missing previous overwrites for #{channel} ({channel.id}); defaulting to None.")
-            overwrite.update(send_messages=None, add_reactions=None)
+            overwrite.update(send_messages=None, add_reactions=None, speak=None)
```

Comment:
They are just ignored (I believe at the d.py level, as `PermissionOverwrite` objects can accept any permissions). Do you think it would be better to add checks here and set the overwrites based on the channel type?

Reply:
No, it's fine if they're being ignored.